nexuse2e-webapp/src/main/frontend/src/app/login/login.component.ts

```diff
@@ -0,0 +1,69 @@
+import { Component, Injectable, OnInit } from "@angular/core";
+import { ActivatedRoute, Router } from "@angular/router";
+import { DataService } from "../data/data.service";
+import { CacheService } from "../data/cache.service";
+import { NavigationService } from "../navigation/navigation.service";
+
+interface LoginData {
+  user: string;
+  password: string;
+}
+
+@Component({
+  selector: "app-login",
+  templateUrl: "./login.component.html",
+  styleUrls: ["./login.component.scss"],
+})
+@Injectable({ providedIn: "root" })
+export class LoginComponent implements OnInit {
+  returnUrl!: string;
+  isHttps?: boolean;
+  machineName?: string;
+  loginError?: boolean;
+  user?: string;
+  password?: string;
+
+  constructor(
+    private route: ActivatedRoute,
+    private router: Router,
+    private dataService: DataService,
+    private cacheService: CacheService,
+    private navigationService: NavigationService
+  ) {}
+
+  async ngOnInit() {
+    try {
+      await this.dataService.get("/logged-in");
+      await this.router.navigate(["/dashboard"]);
+    } catch {
+      const returnUrl = "returnUrl";
+      this.returnUrl = this.route.snapshot.queryParams[returnUrl] || "/";
+      this.isHttps = location.protocol === "https:";
+      this.cacheService
+        .get<string>("/machine-name")
+        .then((name) => (this.machineName = name));
+    }
+  }
+
+  async onSubmit() {
+    if (this.user && this.password) {
+      this.loginError = false;
+      const loginData: LoginData = {
+        user: this.user,
+        password: this.password,
+      };
+
+      try {
+        await this.login(loginData);
+      } catch {
+        this.loginError = true;
+      }
+    }
+  }
+
+  async login(loginData: LoginData) {
+    await this.dataService.post("/login", loginData);
+    this.navigationService.hideNavigation();
+    await this.router.navigateByUrl(this.returnUrl);
```

Comment:
Can we do the returnUrl calcuation directly here? I found it is a little bit spread around this class and I currently do not see why.

Reply:
Made the return url calculation more compact